### PR TITLE
Update bitwarden extension

### PR DESCRIPTION
**Pull request by ivaarsson**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — COMMENTED** [review]
Thanks for your contribution @ivaarsson 🔥 

@jomifepe can you check this?

**Comment by pascalinger** [comment]
Just jumping in as someone affected by the extended fetching times for vault item data: looking forward to seeing this pushed soon to see if it has any effect on my end.

Thank everyone for your work! Really appreciated!

**Comment by ivaarsson** [comment]
I found that I had a vault loading time of approximately 8 seconds which meant it took me 8 seconds to get a password. With this changed that is reduced to roughly 3 seconds.

When I run `bw get item <item id>` in my terminal it takes roughly 3 seconds for the command to finish as well. However, I have noticed that the output for the item is generated in approximately 1 second, yet the command doesn't complete until around 2 seconds later. Although I'm uncertain about the reason behind this delay, I believe there's room for improvement in the bitwarden-cli project.

File: extensions/bitwarden/src/api/bitwarden.ts [hunk 1]
```diff
@@ -363,6 +363,30 @@ export class Bitwarden {
...
+      const { stdout } = await this.exec(["get", "password", id], { resetVaultTimeout: true });
+      return { result: stdout };
+    } catch (execError) {
+      captureException("Failed to get password", execError);
+      const { error } = await this.handleCommonErrors(execError);
+      if (!error) throw execError;
+      return { error };
+    }
+  }
+
```
Comment (jomifepe) [inline extensions/bitwarden/src/api/bitwarden.ts:377]: This function doesn't seem to be used anywhere, can be removed.

```suggestion
```

File: extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts [hunk 2]
```diff
@@ -9,8 +11,19 @@ import { Item } from "~/types/vault";
...
  */
 function useGetUpdatedVaultItem() {
+  const session = useSession();
+  const bitwarden = useBitwarden();
   const getItemFromVault = useVaultItemSubscriber();
 
+  async function getItemNoCache(id: string): Promise<Item> {
+    if (!session.active || !session.token) {
+      throw new Error("No session token available");
+    }
```
Comment (jomifepe) [inline extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts:21]: No need to check this, this is already checked when bootstrapping the session context.

Don't forget to remove the import!

```suggestion
  const bitwarden = useBitwarden();
  const getItemFromVault = useVaultItemSubscriber();

  async function getItemNoCache(id: string): Promise<Item> {
```

**Review by jomifepe — APPROVED** [review]
This is great addition, thanks for implementing it!

Left some comments, but the code looks good 🚀

**Comment by pernielsentikaer** [comment]
Thanks for checking, @ivaarsson could you look into this?

**Comment by ivaarsson** [comment]
@pernielsentikaer The comments should now be resolved

File: extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts [hunk 3]
```diff
@@ -9,8 +10,15 @@ import { Item } from "~/types/vault";
  * Otherwise, it will wait for the value to be retrieved from the vault.
  */
 function useGetUpdatedVaultItem() {
+  const bitwarden = useBitwarden();
   const getItemFromVault = useVaultItemSubscriber();
 
+  async function getItemNoCache(id: string): Promise<Item> {
```
Comment (jomifepe) [inline extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts:16]:
```suggestion
  const waitForItemLoaded = useVaultItemSubscriber();

  async function getItemFromVault(id: string): Promise<Item> {
```

File: extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts [hunk 4]
```diff
@@ -20,7 +28,9 @@ function useGetUpdatedVaultItem() {
     if (!valueHasSensitiveValuePlaceholder(currentValue)) return currentValue;
 
     const toast = loadingMessage ? await showToast(Toast.Style.Animated, loadingMessage) : undefined;
-    const value = selector(await getItemFromVault(possiblyCachedItem.id));
+    const value = selector(
+      await Promise.race([getItemFromVault(possiblyCachedItem.id), getItemNoCache(possiblyCachedItem.id)])
```
Comment (jomifepe) [inline extensions/bitwarden/src/components/searchVault/utils/useGetUpdatedVaultItem.ts:32]:
```suggestion
      await Promise.race([waitForItemLoaded(possiblyCachedItem.id), getItemFromVault(possiblyCachedItem.id)])
```

**Review by jomifepe — COMMENTED** [review]
Small naming improvement

**Comment by pernielsentikaer** [comment]
a few more @ivaarsson - almost ready to merge 🙂

**Comment by ivaarsson** [comment]
> Small naming improvement, my `getItemFromVault` name didn't make much sense

True, accepted these changes as well @pernielsentikaer

**Comment by pernielsentikaer** [comment]
Are you ready for the merge @jomifepe 🙂

**Comment by jomifepe** [comment]
Ship it 🚀

**Pull request merged**